src/js/util/index.js

```diff
@@ -317,7 +318,7 @@ export const getHighlightIndices = (baani, query, type) => {
   let mainQuery = query.replace(/"/g, '');
 
   //Handles - search operator
-  mainQuery = mainQuery.replace(/[-][ ,\w,),(]*/g, '');
+  mainQuery = unicode(mainQuery.replace(/[-][ ,\w,),(]*/g, ''), true);
```

Comment:
@AkalUstat Why are we converting the search query to unicode here? 

We never used anvaad js directly, API was supposed to use that for us until now.

Reply:
Got the answer from your above description. Just thinking if this is the best way to solve it.

What if we don't convert the search query to unicode ever? When switching to unicode, everything else switches, but not the search query.

Reply:
also i really didnt bother to see what was happening with the highlighted index generator i just plugged this in cause it was the first solution i thought of

Reply:
Okay, let me take a look at the highlighted index generator, possibly an error with the regex we use for that. If it's solvable from there, I'd prefer that.

Reply:
any update @Gauravjeetsingh ?

Reply:
I am not able to reproduce this issue on my laptop, and safari on iPhone X

But the query in autocomplete is never Unicode. So this PR fix doesn't affect that. And as per the screenshot in issue #1097, the highlight is off by one word in autocomplete as well. 

I think it's safe to close this PR **without merging**, but the ticket can stay open. I need to investigate more.

Reply:
- this is if the user manually enters unicode

Reply:
is this still an issue?